Mathlib/Topology/Bases.lean

```diff
@@ -54,6 +54,23 @@ More fine grained instances for `FirstCountableTopology`,
 
 open Set Filter Function Topology
 
+namespace Set
+variable {ι : Type*} {α : ι → Type*} {i : ι} {s : Set ι} {t : ∀ i, Set (α i)}
+
+lemma eval_image_pi_of_not_mem [Decidable (s.pi t).Nonempty] (hi : i ∉ s) :
+    Function.eval i '' s.pi t = if (s.pi t).Nonempty then univ else ∅ := by
```

Comment:
This probably belongs to `Data/Set/Prod`.

Reply:
Whoops sorry I indeed meant to put it there